.github/workflows/staging.yml

```diff
@@ -0,0 +1,79 @@
+name: staging-CI
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  build:
+    strategy:
+      matrix:
+        node: [ 13 ]
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: npm install
+      - name: Checking prettied files
+        run: npm run prettier:check
+        # TODO #118: - name: Linting with eslint
+        # TODO #118: run: npm run lint
+      - name: Testing with coverage
+        run: npm test -- --coverage --watchAll=false
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          file: ./coverage/clover.xml
+          flags: unittests
+          name: codecov-umbrella
+          fail_ci_if_error: true
+  deploy_firebase:
+    needs: build
+    name: Deploy to Firebase
+    strategy:
+      matrix:
+        node: [ 13 ]
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: npm install
+      - name: Building produciton version
+        run: npm run build
```

Comment:
I would add the `install` and `build` steps to the build pipeline. Seems more fitting, and these steps can also be executed for the `dev` CI. The tests might pass and yet the build might fail in some rare cases.
Yet I'm not an expert on GitHub actions and maybe you added this here, because resources (e.g. build artifacts) cannot be shared between different steps in the pipeline. The documentation isn't as great so I couldn't research this quickly.

Reply:
The first approach on caching the dependencies from the documentation didn't work so well. It take much more time (15 min) than the clean installation. Maybe I wanna find some performance increase.